src/js/components/Graphs/Legend/legend.spec.jsx

```diff
@@ -14,59 +15,41 @@ import {
-
-    expect(wrapper.contains(expectedValue)).toBe(true);
+  it('Legend comonent is visible and has correctly number of sections', () => {
+    expect(wrapper.getDOMNode()).toBeVisible();
+    expect(legendSections).toHaveLength(4);
   });
 
-  it('Should render legend section and set correctly icon shape and color for labels', () => {
-    const expectedValue = getExpectedValue('Labels', [{ icon: <RectangleIcon color="green" />, desc: '_success' }]);
-    const wrapper = getWrapper('Labels', legendArraysMock.labels);
-
-    expect(wrapper.contains(expectedValue)).toBe(true);
+  it('Sections have correctly number of items, consistent with legendArrays', () => {
+    expect(legendSections.at(0).find(LegendItem)).toHaveLength(legendArrays.nodes.length);
+    expect(legendSections.at(1).find(LegendItem)).toHaveLength(legendArrays.composites.length);
+    expect(legendSections.at(2).find(LegendItem)).toHaveLength(legendArrays.labels.length);
+    expect(legendSections.at(3).find(LegendItem)).toHaveLength(legendArrays.edges.length);
   });
 
-  it('Should render legend section and set correctly icon shape and color for edges', () => {
-    const expectedValue = getExpectedValue('Edges', [
-      { icon: <LineIcon color="black" shape="solid" />, desc: 'processed' },
-      { icon: <LineIcon color="grey" shape="dashed" />, desc: 'unprocessed' },
-    ]);
-    const wrapper = getWrapper('Edges', legendArraysMock.edges);
-
-    expect(wrapper.contains(expectedValue)).toBe(true);
+  it('Every items in section should have correctly icon', () => {
+    expect(legendSections.at(0).find(SquareIcon)).toHaveLength(legendArrays.nodes.length);
+    expect(legendSections.at(1).find(CircleIcon)).toHaveLength(legendArrays.composites.length);
+    expect(legendSections.at(2).find(RectangleIcon)).toHaveLength(legendArrays.labels.length);
+    expect(legendSections.at(3).find(LineIcon)).toHaveLength(legendArrays.edges.length);
   });
 
-  it('Should render legend section and set correctly icon shape and color for undefined elements', () => {
-    const expectedValue = getExpectedValue('Undefined', [{ icon: '', desc: 'undefined' }]);
-    const wrapper = getWrapper('Undefined', legendArraysMock.undefined);
-
-    expect(wrapper.contains(expectedValue)).toBe(true);
+  it('Every items in section should have description', () => {
+    // expect(legendSections.at(0).find(LegendItemDescription)).toHaveLength(legendArrays.nodes.length);
+    legendSections.at(0).find(LegendItemDescription).forEach((node) => {
+      expect(node.text()).not.toEqual('');
+    });
+    legendSections.at(1).find(LegendItemDescription).forEach((node) => {
+      expect(node.text()).not.toEqual('');
+    });
+    legendSections.at(2).find(LegendItemDescription).forEach((node) => {
+      expect(node.text()).not.toEqual('');
+    });
+    legendSections.at(3).find(LegendItemDescription).forEach((node) => {
+      expect(node.text()).not.toEqual('');
```

Comment:
Can this be achieved with `forEach` or it wouldn't be descriptive enough?
@SzajekSzymon @mikolajkozakiewicz